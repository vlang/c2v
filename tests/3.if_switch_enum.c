#include <stdio.h>

typedef enum {
	one, two
} Number;

Number return_number() {
	return one;
}

int main() {
	if (10 > 5) {
		printf("10 > 5");
	}
	else {
		printf("no");
	}
	int x = 10;
	int y = 20;
	if (y > x) {
		printf("y > x");
	}
	if (y + 1 > x); // TODO
	if (1) printf("one");
	else if (1 > 0 || x < y || (x > y && x < y + 1)) {
		printf("two");
		printf("three");
	}
	switch (x) {
		case 0: return 0;
		case 1: printf("one"); printf("ONE"); break;
		case 2: printf("two"); if (1 > 0) { printf("OK"); } break;
	}
	Number n = one;
	switch (n) {
		//case one: printf("one!"); int x = one + 1; break;
			//x := .one + 1
		case one: printf("one"); break;
		case two: printf("two"); break;
	}
	// handle enum <=> int in C: enum switch needs explicit casts to ints
	int m = 0;
	switch (m) {
		case one: printf("one"); break;
		case two: printf("two"); break;
	}
	return 0;
}

