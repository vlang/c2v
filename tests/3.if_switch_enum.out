[translated]
module main

enum Number {
	one
	two
}

fn return_number() Number {
	return Number.one
}

fn main() {
	if 10 > 5 {
		C.printf(c'10 > 5')
	} else {
		C.printf(c'no')
	}
	x := 10
	y := 20
	if y > x {
		C.printf(c'y > x')
	}
	if y + 1 > x {
	}
	if 1 {
		C.printf(c'one')
	} else if 1 > 0 || x < y || (x > y && x < y + 1) {
		C.printf(c'two')
		C.printf(c'three')
	}
	match x {
		0 { // case comp body kind=ReturnStmt is_enum=false
			return
		}
		1 { // case comp body kind=CallExpr is_enum=false
			C.printf(c'one')
			C.printf(c'ONE')
		}
		2 { // case comp body kind=CallExpr is_enum=false
			C.printf(c'two')
			if 1 > 0 {
				C.printf(c'OK')
			}
		}
		else {}
	}
	n := Number.one
	match n {
		.one { // case comp body kind=CallExpr is_enum=true
			C.printf(c'one')
		}
		.two { // case comp body kind=CallExpr is_enum=true
			C.printf(c'two')
		}
		else {}
	}
	m := 0
	match Number(m) {
		.one { // case comp body kind=CallExpr is_enum=true
			C.printf(c'one')
		}
		.two { // case comp body kind=CallExpr is_enum=true
			C.printf(c'two')
		}
		else {}
	}
	return
}
